### PR TITLE
added page to using an algorithm

### PR DESCRIPTION
**Pull request by lennessyy**
- Added a page to explain how to use an algorithm

### CODE REVIEW

File: modules/using-an-algorithm/pages/index.adoc [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+:experimental:
+
+
+This page provides information on how to use a TigerGraph Graph Data Science (GDS) Library algorithm.
+
+All GDS algorithms can be found at https://github.com/tigergraph/gsql-graph-algorithms[`tigergraph/gsql-graph-algorithms`] on GitHub in the `algorithms` folder.
+The algorithms are divided into several categories.
+Each algorithm has its own folder, where you can find the implementation of the algorithm, any user-defined functions required to install the algorithm, or different versions of the algorithm if they exist.
+
+Once you have decided that an algorithm fits your use case, it's time to run the algorithm on your graph data.
```
Comment (victorleeTG) [inline modules/using-an-algorithm/pages/index.adoc]: This sentence would fit more in a walk-through tutorial than in here, which I think is closer to a reference guide.

File: modules/using-an-algorithm/pages/index.adoc [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+For example, xref:classification-algorithms:k-nearest-neighbors.adoc[] runs on graphs with either directed or undirected edges, but the edges must have a weight attribute.
+
+Another example is the xref:node-embeddings:fast-random-projection.adoc[Fast Random Projection algorithm], which expects the vertex type to have an attribute of type `LIST<DOUBLE>` if you want to store the embedding results to your graph data.
+
+== Install necessary UDFs
+
+Some algorithms require the use of user-defined functions (UDF).
+The UDF files are included in the same folder as the algorithm itself.
+
+If your query requires a UDF, ensure that the UDF files are added to the GSQL before creating or installing the query.
```
Comment (victorleeTG) [inline modules/using-an-algorithm/pages/index.adoc]: 'If your query requires a UDF, ensure that the UDF files are added to the GSQ'

I would reword this:
'If your query requires a UDF, ensure that the UDF has been installed into your TigerGraph system'

File: modules/using-an-algorithm/pages/index.adoc [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
++
+--
+. Locate the query in the GDS Library GitHub repository.
+It is a `.gsql` file named after the query.
+. Copy the entire content of the query file.
+This is the command to create the query.
+. Log in to the GSQL shell as a user with query writing privileges for the graph on which you want to create the query.
+. Run `USE GRAPH <graph_name>` to switch to the graph you want to create the query on.
+. Type `BEGIN` to start a multi-line command.
+. Paste the `CREATE QUERY` command into the terminal.
```
Comment (victorleeTG) [inline modules/using-an-algorithm/pages/index.adoc]: I would not recommend this method.  I think it is much better to have the query in a file, and then to have GSQL read the file.

File: modules/using-an-algorithm/pages/index.adoc [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+Another example is the xref:node-embeddings:fast-random-projection.adoc[Fast Random Projection algorithm], which expects the vertex type to have an attribute of type `LIST<DOUBLE>` if you want to store the embedding results to your graph data.
+
+== Install necessary UDFs
+
+Some algorithms require the use of user-defined functions (UDF).
+The UDF files are included in the same folder as the algorithm itself.
+
+If your query requires a UDF, ensure that the UDF files are added to the GSQL before creating or installing the query.
+If your query does not require a UDF, you can proceed to create the query.
```
Comment (victorleeTG) [inline modules/using-an-algorithm/pages/index.adoc]: We are working towards having the algorithms already included in the product. So, rather than saying that "The next step is to create the query", I'd rather say something like
"If the algorithm you wish to use is not yet installed in your TigerGraph instance, you need to install it."
Then explain what are the currently available ways to do that.
The old documentation gave the basic method.
The TigerCLI and the upcoming GraphStudio feature will provide 2 additional ways.

File: modules/using-an-algorithm/pages/index.adoc [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
++
+--
+WARNING: Saving a query in GraphStudio does not create the query in GSQL.
+
+. Locate the query in the GDS Library GitHub repository.
+It is a `.gsql` file named after the query.
+. Copy the entire content of the query file.
+This is the command to create the query.
+. Log in to GraphStudio as a user with query writing privileges for the graph on which you want to create the query.
+. Click btn:[Global View] in the top-left corner and choose the graph to use.
```
Comment (victorleeTG) [inline modules/using-an-algorithm/pages/index.adoc:69]: Why are we abbreviating button?

**Review by victorleeTG — COMMENTED** [review]
I thought I wouldn't have any comments, but I was wrong :-)

File: modules/using-an-algorithm/pages/index.adoc (hunk 1 above)
Comment (lennessyy) [reply modules/using-an-algorithm/pages/index.adoc]: I wrote this page as a how-to, as this is about how to use an algorithm. 
I was originally thinking that picking an algorithm is part of using one because we started by describing how to find an algorithm from the library, but in the end, this sentence does seem redundant. I deleted it.

File: modules/using-an-algorithm/pages/index.adoc (hunk 5 above)
Comment (lennessyy) [reply modules/using-an-algorithm/pages/index.adoc:69]: btn:[] is called a macro in Asciidoc. Basically this applies a style to the texts within to make it clear that the texts represent a button. `btn:[Global View]` looks like **[ Global View ]** when rendered right now. In the future if we want to change the style, we will be able to change the style of all buttons by editing the CSS.

File: modules/using-an-algorithm/pages/index.adoc [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+
+== Check schema/data constraints
+Most algorithm queries in the GDS Library are schema-free, meaning that you are able to run the query on any schema.
+However, some algorithms have certain schema or data constraints by nature.
+Make sure to read the documentation for the algorithm to determine the following:
+
+* Does the algorithm require edges to be directed/undirected?
+* Does the algorithm require edges to be weighted/unweighted?
+* Does the algorithm require any vertex type to have an attribute of a certain data type?
+* Does the algorithm expect your data to have been processed in a certain way before it runs?
```
Comment (alexthomasTG) [inline modules/using-an-algorithm/pages/index.adoc]: Should "expect" be "require" here as well?

File: modules/using-an-algorithm/pages/index.adoc (hunk 2 above)
Comment (alexthomasTG) [reply modules/using-an-algorithm/pages/index.adoc]: We can also link here to the UDF page.

File: modules/using-an-algorithm/pages/index.adoc [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+Installing a query allows the algorithm query to access all features offered by the GSQL Query language.
+It also increases the performance of the query.
+
+To install a query, run `INSTALL QUERY <query_name>` in the GSQL shell.
+Alternatively, you can click btn:[Install] on the btn:[Write Query] page of GraphStudio.
+
+=== Install query in distributed mode
+If you are running the query on a TigerGraph cluster, you may consider installing the query in distributed mode.
+
+In general, distributed mode is likely to improve the performance of a query if the query meets the following:
```
Comment (alexthomasTG) [inline modules/using-an-algorithm/pages/index.adoc]: meets either of the following conditions:

File: modules/using-an-algorithm/pages/index.adoc (hunk 4 above)
Comment (lennessyy) [reply modules/using-an-algorithm/pages/index.adoc]: I was thinking whenever our algorithm update comes out, we'll update this section accordingly. For now since it isn't out, we describe the steps necessary as they are now.

File: modules/using-an-algorithm/pages/index.adoc (hunk 2 above)
Comment (lennessyy) [reply modules/using-an-algorithm/pages/index.adoc]: Updated.

File: modules/using-an-algorithm/pages/index.adoc (hunk 3 above)
Comment (lennessyy) [reply modules/using-an-algorithm/pages/index.adoc]: Updated to paste into file and invoke from GSQL.
